doc/mgmt/SONiC_AliasFeature_HLD.md

```diff
@@ -0,0 +1,337 @@
+
+   **Sub-tree Transformer**: In cases where applications use subtree transformer, then need to use an API that will return the alias string. The API will internally figure out if alias is enabled or not.
+
+   Following are the APIs used:  
+   These APIs are available in `translib/utils` package.
+
+   **GetNativeNameFromUIName**  
+   `func GetNativeNameFromUIName(ifName *string) *string`  
+
+   Retrieves native interface name from user input name, if the alias mode is enabled. User input name can be alias or native interface. API will provide native interface name if the user input is alias name, otherwise it will return the native interface name passed to the API. If the alias mode is not enabled, API will return the input string passed.
+
+   **GetAliasNameFromUIName**  
+   `func GetAliasNameFromUIName(ifName *string) *string`  
+
+   Retrieves alias interface name from user input name, if the alias mode is enabled. User input name can be alias or native interface. API will provide alias name if the user input is native interface name, otherwise it will return the alias name passed to the API. If the alias mode is not enabled, API will return the input string passed.
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+
+#### 3.2.1.1
+
+### 3.2.2 APP DB
+
+### 3.2.3 STATE DB
+
+### 3.2.4 ASIC DB
+
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+
+### 3.3.1 Orchestration Agent
+
+### 3.3.2 Other Process
+
+## 3.4 SyncD
+
+## 3.5 SAI
+
+## 3.6 User Interface
+
+### 3.6.1 Data Models
+Yang model used for alias mode handling  
+[sonic-device-metadata.yang](https://github.com/project-arlo/sonic-mgmt-framework/blob/master/models/yang/sonic/sonic-device-metadata.yang)  
+
+Supported yang objects and attributes are highlighted in green:
+```diff
+module: sonic-device-metadata
+    +--rw sonic-device-metadata
+       +--rw DEVICE_METADATA
+          +--rw DEVICE_METADATA_LIST* [name]
+             +--rw name                          string
++            +--rw aliasMode                     boolean
+```
+
+### 3.6.2 CLI
+
+#### 3.6.2.1 Configuration Commands
+
+#### Alias mode configuration
+
+`[no] alias enable`
+```
+sonic(config)# alias enable
+sonic(config)# no alias enable
+```
+#### Config commands when alias mode is enabled
+`Eth 1/2`, `Eth1/2`, and `e1/2` options are supported to get into alias interface config mode.  The parser converts all to `Eth1/2`.

```

Comment:
This means that there is no long or short name for interface i.e Ethernet1/1 will be rejected in config commands (all ui type) and will not be part of any show commands

Reply:
Short form is only when alias is enabled. Native mode will have long form. This has been discussed & agreed by both teams earlier.